### DISCUSSION TITLE
Disable console message to use @tensorflow/tfjs-node

### DISCUSSION THREAD

**Discussion by tarounen** [vladmandic/face-api#58] [Q&A] [answered]
Hello, I am trying to check face similarity using face-api on google cloud function. I was not a able to install @tensorflow/tfjs-node on my windows machine. I keep getting some error on install.  Ended up using:

 `const faceapi = require("@vladmandic/face-api/dist/face-api.node-cpu.js");`

which is working fine. However I keep getting the console message: 

```
  ============================
 Hi there �. Looks like you are running TensorFlow.js in Node.js. To speed things up dramatically, install our node backend, which 
 binds to TensorFlow C++, by running npm i @tensorflow/tfjs-node, or npm i @tensorflow/tfjs-node-gpu if you have CUDA. 
 Then call require('@tensorflow/tfjs-node'); (-gpu suffix for CUDA) at the start of your program. 
 Visithttps://github.com/tensorflow/tfjs-node for more details.
 ============================
```

which will increase the logs on cloud functions. Is there an option to suppress this message?

**Comment by vladmandic** [comment, marked as answer]
unfortunately, no.
the message comes from within tfjs-core package and there is no way to suppress it. but it would be a good suggestion to create a way to suppress it - i suggest to open an issue there: <https://github.com/tensorflow/tfjs/issues>

**Reply by tarounen** [reply]
Thank you for the response. 

Actually I am facing another issue. using  `faceapi.computeFaceDescriptor` turns out to be not accurate, whenever I try to compare a picture with a face and a picture with no face, most the of the time the distance is < 0.6 (hence a match). 

I tried to solve this by going a `faceapi.detectSingleFace` first, then checks if any face is present then proceed with the `computerFaceDescriptor`. However I had to drop this idea as its taking too much time to run on Cloud Functions and is resulting in a timeout. 

The next solution I am trying is to detect the face on front end (vuejs) first, if there is faces in both photos, then call the cloud function to check if their similarity.  But im stuck with a `Could not get context for WebGL version 2` upon loading the model. so it ends up failing when calling `detectSingleFace`. Find the implementation and log for this one below (i purposely remoce the img src): 

Import: 

```
import * as tf from '@tensorflow/tfjs';
import * as canvas from 'canvas';
import * as faceapi from '@vladmandic/face-api/dist/face-api.node-cpu.js';
```

Code: 

```
const MODEL_URL = "/models";

await this.$faceapi.nets.ssdMobilenetv1.loadFromUri(MODEL_URL);
await this.$faceapi.nets.faceRecognitionNet.loadFromUri(MODEL_URL);
await this.$faceapi.nets.faceLandmark68Net.loadFromUri(MODEL_URL);

console.log("loaded");

const photo1 = await this.$canvas.loadImage(this.photo1Link);
const photo2 = await this.$canvas.loadImage(this.photo2Link);

console.log(photo1);
console.log(photo2);

const result = await this.$faceapi.detectSingleFace(photo1);
if (!result) {
	console.error("error: NO FACE DETECTED in photo1");
	return;
}
```
Console: 

```
Could not get context for WebGL version 2
loaded
<img src="photo1url">
<img src="photo2url">
gpgpu_util.js?ce57:124 Uncaught (in promise) DOMException: Failed to execute 'texImage2D' on 'WebGLRenderingContext': Tainted canvases may not be loaded.
at eval (webpack-internal:///./node_modules/@tensorflow/tfjs-backend-webgl/dist/gpgpu_util.js:149:79)
at Module.callAndCheck (webpack-internal:///./node_modules/@tensorflow/tfjs-backend-webgl/dist/webgl_util.js:67:25)
at Module.uploadPixelDataToTexture (webpack-internal:///./node_modules/@tensorflow/tfjs-backend-webgl/dist/gpgpu_util.js:149:65)
at GPGPUContext.uploadPixelDataToTexture (webpack-internal:///./node_modules/@tensorflow/tfjs-backend-webgl/dist/gpgpu_context.js:132:77)
at Object.fromPixels [as kernelFunc] (webpack-internal:///./node_modules/@tensorflow/tfjs-backend-webgl/dist/kernels/FromPixels.js:62:19)
at kernelFunc (webpack-internal:///./node_modules/@tensorflow/tfjs-core/dist/engine.js:478:30)
at eval (webpack-internal:///./node_modules/@tensorflow/tfjs-core/dist/engine.js:539:27)
at Engine.scopedRun (webpack-internal:///./node_modules/@tensorflow/tfjs-core/dist/engine.js:352:25)
at Engine.runKernelFunc (webpack-internal:///./node_modules/@tensorflow/tfjs-core/dist/engine.js:535:14)
at Engine.runKernel (webpack-internal:///./node_modules/@tensorflow/tfjs-core/dist/engine.js:408:21)
```

**Reply by vladmandic** [reply]
error `Could not get context for WebGL version 2` is because browser does not allow both 2D context and WebGL context to be active for the same canvas - which again means you're using 2D context in your function, likely within `this.$canvas.loadImage(this.photo1Link);`

You should either simplify that function somehow not to use 2d context or add one more canvas copy so there is a canvas without context before passing it on for detection.
